qa/tasks/cephfs/test_admin.py

```diff
@@ -2659,3 +2659,28 @@ def test_with_health_warn_with_2_active_MDSs(self):
                               errmsgs=health_warn)
         self.run_ceph_cmd(f'mds fail {mds1_id} --yes-i-really-mean-it')
         self.run_ceph_cmd(f'mds fail {mds2_id} --yes-i-really-mean-it')
+
+
+class TestFSSet(TestAdminCommands):
+
+    def test_fs_set_max_mds_when_unhealthy_without_confirm(self):
+        '''
+        Test that command "ceph fs set max_mds <num>" without the confirmation
+        flag (--yes-i-really-mean-it) fails when cluster is unhealthy.
+        '''
+        self.gen_health_warn_mds_cache_oversized()
+
+        with self.assertRaises(CommandFailedError) as cfe:
+            self.fs.set_max_mds(2)
+        self.assertEqual(cfe.exception.exitstatus, errno.EPERM)
+
+    def test_fs_set_max_mds_when_unhealthy_with_confirm(self):
+        '''
+        Test that command "ceph fs set max_mds <num> --yes-i-really-mean-it"
+        run successfully when cluster is unhealthy.
+        '''
+        self.gen_health_warn_mds_cache_oversized()
+
+        self.fs.set_max_mds(2, confirm=True)
```

Comment:
You could also write a test case using `confirm=True` when the cluster is healthy and it passes.

Reply:
That's a good idea. I've added it.